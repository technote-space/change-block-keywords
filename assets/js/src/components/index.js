import Keywords from './keywords';

export { Keywords };
