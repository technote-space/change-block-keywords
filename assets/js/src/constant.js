export const PLUGIN_NAME = 'change-block-keywords';
export const TRANSLATE_DATA = cbkParams.translate;
/** @var {{ translate: string[] }} cbkParams */