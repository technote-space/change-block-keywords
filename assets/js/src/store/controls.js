/**
 * @param {string} storeKey store key
 * @param {string} selectorName selector name
 * @param {object} args args
 * @returns {{args: *[], selectorName: *, storeKey: *, type: string}} control
 */
export const select = ( storeKey, selectorName, ...args ) => ( {
	type: 'SELECT',
	storeKey,
	selectorName,
	args,
} );

export default {
	SELECT( { storeKey, selectorName, args } ) {
		return wp.data.select( storeKey )[ selectorName ]( ...args );
	},
};
