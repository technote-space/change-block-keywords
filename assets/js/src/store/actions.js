import { STORE_NAME, SET_KEYWORDS, RESET_KEYWORD, INITIALIZED } from './constant';
import {
	select,
} from './controls';

/**
 * @param {string} name name
 * @param {array} keywords keywords
 */
export function* initialize( name, keywords ) {
	if ( ! ( yield select( STORE_NAME, 'hasInitialized', name ) ) ) {
		yield resetKeyword( name, keywords );
	}
}

/**
 * @param {string} name name
 * @param {array} keywords keywords
 */
export function* setKeywords( name, keywords ) {
	yield {
		type: SET_KEYWORDS,
		name,
		keywords,
	};
}

/**
 * @param {string} name name
 * @param {array} keywords keywords
 */
export function* resetKeyword( name, keywords ) {
	if ( keywords ) {
		yield {
			type: RESET_KEYWORD,
			name,
			keywords: keywords,
		};
	}
	yield {
		type: INITIALIZED,
		name,
	};
}
