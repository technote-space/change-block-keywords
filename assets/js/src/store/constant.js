import { PLUGIN_NAME } from '../constant';

export const STORE_NAME = PLUGIN_NAME;

export const SET_KEYWORDS = '@@CBK/SET_KEYWORDS';
export const RESET_KEYWORD = '@@CBK/RESET_KEYWORD';
export const INITIALIZED = '@@CBK/INITIALIZED';
export const RESET_ALL = '@@CBK/RESET_ALL';
