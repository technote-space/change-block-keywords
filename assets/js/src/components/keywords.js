const { FormTokenField } = wp.components;
const { InspectorAdvancedControls } = wp.editor;
const { compose } = wp.compose;
const { withSelect, withDispatch } = wp.data;

import { STORE_NAME } from '../store/constant';
import { translate } from '../utils';

/**
 * @param {object} props props
 * @param {function} keywords keywords
 * @param {function} setKeywords set keywords
 * @returns {Component} input keywords form
 * @constructor
 */
export function Keywords( { props, keywords, setKeywords } ) {
	return <InspectorAdvancedControls>
		<FormTokenField
			value={ keywords( props ) }
			onChange={ tokens => setKeywords( props, tokens ) }
			label={ translate( 'Set Search Keywords' ) }
		/>
	</InspectorAdvancedControls>;
}

export default compose(
	withSelect( select => {
		return {
			keywords: props => select( STORE_NAME ).getKeywords( props.name ),
		};
	} ),
	withDispatch( dispatch => {
		return {
			setKeywords: ( props, keywords ) => dispatch( STORE_NAME ).setKeywords( props.name, keywords ),
		};
	} ),
)( Keywords );
