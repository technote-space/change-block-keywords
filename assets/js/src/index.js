const { addFilter } = wp.hooks;

import './store';
import { getNamespace, getKeywordsFormComponent, getSetupKeywordsFunc } from './utils';

addFilter(
	'editor.BlockEdit',
	getNamespace( 'add-keywords-form' ),
	getKeywordsFormComponent(),
);

addFilter(
	'blocks.registerBlockType',
	getNamespace( 'setup-keywords' ),
	getSetupKeywordsFunc(),
);
