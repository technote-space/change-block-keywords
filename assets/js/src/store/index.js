import { registerStore } from '@wordpress/data';
import { STORE_NAME } from './constant';
import reducer from './reducer';
import * as selectors from './selectors';
import * as actions from './actions';
import controls from './controls';

const store = registerStore(STORE_NAME, { reducer, selectors, actions, controls, persist: ['hasInitialized', 'keywords'] });
export default store;
