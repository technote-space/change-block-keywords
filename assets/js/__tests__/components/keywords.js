/* eslint-disable no-magic-numbers */
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Keywords } from '../../src/components';
import { STORE_NAME } from '../../src/store/constant';
import '../../src/store';

const { BlockEdit, InspectorAdvancedControls } = wp.blockEditor;
const { SlotFillProvider } = wp.components;
const { createHigherOrderComponent } = wp.compose;
const { Fragment } = wp.element;
const { addFilter, removeFilter } = wp.hooks;
const { create } = wp.richText;
const { select } = wp.data;
const { ENTER } = wp.keycodes;

beforeAll( () => {
	addFilter( 'editor.BlockEdit', 'components-test/keywords-test', createHigherOrderComponent( BlockEdit => props =>
		<Fragment>
			<BlockEdit { ...props }/>
			<Keywords props={ props }/>
		</Fragment>,
	) );
} );

afterAll( () => {
	removeFilter( 'editor.BlockEdit', 'components-test/keywords-test' );
} );

describe( 'Keywords', () => {
	it( 'should render Keywords', () => {
		const wrapper = mount(
			<SlotFillProvider>
				<InspectorAdvancedControls.Slot/>

				<BlockEdit
					name="core/paragraph"
					isSelected={ true }
					attributes={ ( {
						className: 'test-block-edit',
						content: create( {
							text: 'test',
							start: 0,
							end: 1,
							formats: [ [], [], [], [] ],
						} ),
					} ) }
				/>
			</SlotFillProvider>,
		);

		expect( toJson( wrapper, { mode: 'deep' } ) ).toMatchSnapshot( 'render' );

		expect( select( STORE_NAME ).getKeywords( 'core/paragraph' ) ).toHaveLength( 0 );
		expect( wrapper.find( '.components-form-token-field__input-container' ).hostNodes() ).toHaveLength( 1 );
		expect( wrapper.find( '.components-form-token-field__input' ).hostNodes() ).toHaveLength( 1 );
		wrapper.find( '.components-form-token-field__input' ).hostNodes().simulate( 'change', { target: { value: 'test' } } );
		wrapper.find( '.components-form-token-field' ).hostNodes().simulate( 'keydown', { keyCode: ENTER } );

		expect( toJson( wrapper, { mode: 'deep' } ) ).toMatchSnapshot( 'input' );
		expect( wrapper.find( '.components-form-token-field__token-text' ).hostNodes() ).toHaveLength( 1 );
		expect( select( STORE_NAME ).getKeywords( 'core/paragraph' ) ).toHaveLength( 1 );
		expect( select( STORE_NAME ).getKeywords( 'core/paragraph' ) ).toContain( 'test' );
	} );
} );
