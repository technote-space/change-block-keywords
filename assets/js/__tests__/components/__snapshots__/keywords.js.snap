// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Keywords should render Keywords: input 1`] = `
Array [
  <div
    className="components-form-token-field"
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyPress={[Function]}
    tabIndex="-1"
  >
    <label
      className="components-form-token-field__label"
      htmlFor="components-form-token-input-0"
    >
      Set Search Keywords
    </label>
    <div
      className="components-form-token-field__input-container"
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      tabIndex="-1"
    >
      <span
        className="components-form-token-field__token"
      >
        <span
          className="components-form-token-field__token-text"
          id="components-form-token-field__token-text-0"
        >
          <span
            className="screen-reader-text"
          >
            test (1 of 1)
          </span>
          <span
            aria-hidden="true"
          >
            test
          </span>
        </span>
        <ForwardRef(IconButton)
          aria-describedby="components-form-token-field__token-text-0"
          className="components-form-token-field__remove-token"
          icon="dismiss"
          label="Remove item"
          onClick={[Function]}
        >
          <ForwardRef(Button)
            aria-describedby="components-form-token-field__token-text-0"
            aria-label="Remove item"
            className="components-icon-button components-form-token-field__remove-token"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <button
              aria-describedby="components-form-token-field__token-text-0"
              aria-label="Remove item"
              className="components-button components-icon-button components-form-token-field__remove-token"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              type="button"
            >
              <svg
                aria-hidden="true"
                className="dashicon dashicons-dismiss"
                focusable="false"
                height={20}
                role="img"
                viewBox="0 0 20 20"
                width={20}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M10 2c4.42 0 8 3.58 8 8s-3.58 8-8 8-8-3.58-8-8 3.58-8 8-8zm5 11l-3-3 3-3-2-2-3 3-3-3-2 2 3 3-3 3 2 2 3-3 3 3z"
                />
              </svg>
            </button>
          </ForwardRef(Button)>
        </ForwardRef(IconButton)>
      </span>
      <input
        aria-autocomplete="list"
        aria-describedby="components-form-token-suggestions-howto-0"
        aria-expanded={false}
        className="components-form-token-field__input"
        disabled={false}
        id="components-form-token-input-0"
        onBlur={[Function]}
        onChange={[Function]}
        role="combobox"
        size={1}
        type="text"
        value=""
      />
    </div>
    <p
      className="components-form-token-field__help"
      id="components-form-token-suggestions-howto-0"
    >
      Separate with commas or the Enter key.
    </p>
  </div>,
  Array [
    <div>
       
      Array [
        "",
        "",
        <div
          className="editor-rich-text block-editor-rich-text"
        >
          <div
            onFocus={[Function]}
          >
            <div
              onBlur={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
            >
              <div
                className="components-autocomplete"
                onClick={[Function]}
              >
                <p
                  aria-activedescendant={null}
                  aria-label="Paragraph block"
                  aria-multiline={true}
                  aria-owns={null}
                  className="rich-text editor-rich-text__editable block-editor-rich-text__editable wp-block-paragraph test-block-edit"
                  contentEditable={true}
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "﻿<span data-rich-text-placeholder=\\"Start writing or type / to choose a block\\" contenteditable=\\"false\\"></span>",
                    }
                  }
                  onBlur={[Function]}
                  onCompositionEnd={[Function]}
                  onFocus={[Function]}
                  onInput={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseUp={[Function]}
                  onPaste={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}
                  role="textbox"
                  style={
                    Object {
                      "backgroundColor": undefined,
                      "color": undefined,
                      "direction": undefined,
                      "fontSize": undefined,
                      "whiteSpace": "pre-wrap",
                    }
                  }
                  suppressContentEditableWarning={true}
                />
              </div>
            </div>
          </div>
        </div>,
      ]
       
    </div>,
    "",
  ],
]
`;

exports[`Keywords should render Keywords: render 1`] = `
Array [
  <div
    className="components-form-token-field"
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyPress={[Function]}
    tabIndex="-1"
  >
    <label
      className="components-form-token-field__label"
      htmlFor="components-form-token-input-0"
    >
      Set Search Keywords
    </label>
    <div
      className="components-form-token-field__input-container"
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      tabIndex="-1"
    >
      <input
        aria-autocomplete="list"
        aria-describedby="components-form-token-suggestions-howto-0"
        aria-expanded={false}
        className="components-form-token-field__input"
        disabled={false}
        id="components-form-token-input-0"
        onBlur={[Function]}
        onChange={[Function]}
        role="combobox"
        size={1}
        type="text"
        value=""
      />
    </div>
    <p
      className="components-form-token-field__help"
      id="components-form-token-suggestions-howto-0"
    >
      Separate with commas or the Enter key.
    </p>
  </div>,
  Array [
    <div>
       
      Array [
        "",
        "",
        <div
          className="editor-rich-text block-editor-rich-text"
        >
          <div
            onFocus={[Function]}
          >
            <div
              onBlur={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
            >
              <div
                className="components-autocomplete"
                onClick={[Function]}
              >
                <p
                  aria-activedescendant={null}
                  aria-label="Paragraph block"
                  aria-multiline={true}
                  aria-owns={null}
                  className="rich-text editor-rich-text__editable block-editor-rich-text__editable wp-block-paragraph test-block-edit"
                  contentEditable={true}
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "﻿<span data-rich-text-placeholder=\\"Start writing or type / to choose a block\\" contenteditable=\\"false\\"></span>",
                    }
                  }
                  onBlur={[Function]}
                  onCompositionEnd={[Function]}
                  onFocus={[Function]}
                  onInput={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseUp={[Function]}
                  onPaste={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}
                  role="textbox"
                  style={
                    Object {
                      "backgroundColor": undefined,
                      "color": undefined,
                      "direction": undefined,
                      "fontSize": undefined,
                      "whiteSpace": "pre-wrap",
                    }
                  }
                  suppressContentEditableWarning={true}
                />
              </div>
            </div>
          </div>
        </div>,
      ]
       
    </div>,
    "",
  ],
]
`;
